src/components/environment-map.js

```diff
@@ -0,0 +1,46 @@
+import cubeMapPosX from "../assets/images/cubemap/posx.jpg";
+import cubeMapNegX from "../assets/images/cubemap/negx.jpg";
+import cubeMapPosY from "../assets/images/cubemap/posy.jpg";
+import cubeMapNegY from "../assets/images/cubemap/negy.jpg";
+import cubeMapPosZ from "../assets/images/cubemap/posz.jpg";
+import cubeMapNegZ from "../assets/images/cubemap/negz.jpg";
+
+async function loadEnvMap() {
+  const urls = [cubeMapPosX, cubeMapNegX, cubeMapPosY, cubeMapNegY, cubeMapPosZ, cubeMapNegZ];
+  const texture = await new THREE.CubeTextureLoader().load(urls);
+  texture.format = THREE.RGBFormat;
+  return texture;
+}
+
+AFRAME.registerComponent("environment-map", {
+  schema: {
+    source: { default: "skybox", type: "string" }
```

Comment:
Its a bit weird that nothing ever actually looks for `skybox`, the only meaningful value for source is `defaultCubemap` and that even gets overwritten if there is a skybox. Maybe this should be nothing by default, then the skybox component would set it to `skybox` and then in `environment-map`'s `update` you could grab the skybox cubemap? 

Not too sure, but the way its split up now is a bit confusing.

Reply:
How about just changing this to a boolean `loadDefault` and default it to true. We'd then set it to false in hub.html and scene.html.